File: backend/src/main/java/com/back/domain/shelter/dto/response/ShelterResponseDto.java
package com.back.domain.shelter.dto.response;


import com.back.domain.shelter.entity.Shelter;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record ShelterResponseDto(
        Long id,
        String name,
        String address,
        String city,
        String state,
        String zipCode,
        String phone,
        LocalDateTime createdAt
) {
    public static ShelterResponseDto from(Shelter shelter) {
        return ShelterResponseDto.builder()
                .id(shelter.getId())
                .name(shelter.getName())
                .address(shelter.getAddress())
                .city(shelter.getCity())
                .state(shelter.getState())
                .zipCode(shelter.getZipCode())
                .phone(shelter.getPhone())
                .createdAt(shelter.getCreatedAt())
                .build();
    }
}

File: backend/src/main/java/com/back/domain/shelter/dto/response/ShelterResponseDto.kt
package com.back.domain.shelter.dto.response

import java.time.LocalDateTime

data class ShelterResponseDto(
    val id: Long,
    val name: String,
    val address: String?,
    val city: String?,
    val state: String?,
    val zipCode: String?,
    val phone: String?,
    val createdAt: LocalDateTime
)